op-program/client/interop/interop.go

```diff
@@ -72,7 +76,22 @@ func stateTransition(logger log.Logger, bootInfo *boot.BootInfoInterop, l1Preima
 			return common.Hash{}, err
 		}
 		expectedPendingProgress = append(expectedPendingProgress, block)
+	} else if transitionState.Step == ConsolidateStep {
+		// sanity check
+		if len(transitionState.PendingProgress) >= ConsolidateStep {
```

Comment:
Why `>=` and not `==`?

Reply:
It's to avoid the case where, for example, given a two-sized dependency set, `[Valid_A, Valid_B]` would be equivalent to `[Valid_A, Valid_B, Invalid_C, Invalid_D]`.

There's nothing stopping a challenger from posting a claim with the `>=` case. Without the `>=` check, so long as each pending state per state is valid, the op-program will happily verify the claim. This wouldn't mean the fault proof is unsound, since the first two pending states are indeed valid. But it would be weird to treat both cases the same in the fault proof.

Reply:
Resolving this comment unless you have any objection.